R/settings.R

```diff
@@ -94,54 +94,59 @@ read_config_file <- function(config_file) {
     return(NULL)
   }
 
+  # clickable link for eventual error messages.
+  config_link <- cli::style_hyperlink( # nolint: object_usage_linter. TODO(#2252).
+    cli::col_blue("config"),
+    paste0("file://", config_file)
+  )
   config <- new.env()
   if (endsWith(config_file, ".R")) {
     load_config <- function(file) sys.source(file, config, keep.source = FALSE, keep.parse.data = FALSE)
     malformed <- function(e) {
-      cli_abort(c(
-        "Malformed config file, ensure it is valid R syntax.",
-        conditionMessage(e)
-      ))
+      cli_abort(
+        "Malformed {config_link} file, ensure it is valid R syntax.",
+        parent = e
+      )
     }
   } else {
     load_config <- function(file) {
       dcf_values <- read.dcf(file, all = TRUE)
       for (setting in names(dcf_values)) {
-        parsed_setting <- tryCatch(
+        parsed_setting <- withCallingHandlers(
```

Comment:
why `withCallingHandlers()` over `tryCatch()`?

Reply:
Prodcues better stack trace? I don't mind reverting

Reply:
For non-error signals the difference is clear, agree it's more ambiguous the difference between `tryCatch(error=...)` and `withCallingHandlers(error=...)`.

Since the stack is likely to be helpful here, I agree to use `withCallingHandlers()`.